File: mvvm-java/ui/src/main/java/com/wasisto/githubuserfinder/android/ui/search/SearchHistoryAdapter.java
package com.wasisto.githubuserfinder.android.ui.search;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import com.wasisto.githubuserfinder.android.databinding.ItemSearchHistoryBinding;
import com.wasisto.githubuserfinder.domain.models.SearchHistoryItem;

import java.util.List;

public class SearchHistoryAdapter extends RecyclerView.Adapter<SearchHistoryAdapter.ViewHolder> {

    public static class ViewHolder extends RecyclerView.ViewHolder {

        private ItemSearchHistoryBinding binding;

        public ViewHolder(ItemSearchHistoryBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(SearchHistoryItem historyItem, SearchViewModel viewModel) {
            binding.setHistoryItem(historyItem);
            binding.setViewModel(viewModel);
        }
    }

    private SearchViewModel viewModel;

    private List<SearchHistoryItem> data;

    public SearchHistoryAdapter(SearchViewModel viewModel) {
        this.viewModel = viewModel;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(ItemSearchHistoryBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(SearchHistoryAdapter.ViewHolder holder, int position) {
        holder.bind(data.get(position), viewModel);
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public List<SearchHistoryItem> getData() {
        return data;
    }

    public void setData(List<SearchHistoryItem> data) {
        this.data = data;
    }
}

File: mvvm-kotlin/ui/src/main/java/com/wasisto/githubuserfinder/android/ui/search/SearchHistoryAdapter.kt
package com.wasisto.githubuserfinder.android.ui.search

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.wasisto.githubuserfinder.android.databinding.ItemSearchHistoryBinding
import com.wasisto.githubuserfinder.domain.models.SearchHistoryItem

class SearchHistoryAdapter(private val viewModel: SearchViewModel) : RecyclerView.Adapter<SearchHistoryAdapter.ViewHolder>() {

    class ViewHolder(private val binding: ItemSearchHistoryBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(historyItem: SearchHistoryItem, viewModel: SearchViewModel) {
            binding.historyItem = historyItem
            binding.viewModel = viewModel
        }
    }

    lateinit var data: List<SearchHistoryItem>

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(ItemSearchHistoryBinding.inflate(LayoutInflater.from(parent.context), parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(data[position], viewModel)
    }

    override fun getItemCount(): Int {
        return data.size
    }
}